dbschema/seeds/011.product.edgeql

```diff
@@ -0,0 +1,1080 @@
+                                      book := <str>verseRange['end']['book'],
+                                      chapter := <int16>verseRange['end']['chapter'],
+                                      verse := <int16>verseRange['end']['verse'],
+                                      verseId := <int16>verseRange['end']['verseId']
+                                  }
+                              )
+                          }
+                      )
+                  )
+              })
+              else {},
+            engagement := assert_single((select Engagement filter .project.name = <str>directScriptureProduct['project']))
+        insert DirectScriptureProduct {
+          project := engagement.project,
+          projectContext := engagement.projectContext,
+          engagement := engagement,
+          scripture := scripture,
+          unspecifiedScripture := if (select exists json_get(directScriptureProduct,'unspecifiedScripture')) then (
+              insert Scripture::UnspecifiedPortion {
+                    book := <str>(json_get(directScriptureProduct, 'unspecifiedScripture', 'book')),
+                    totalVerses := <int16>(json_get(directScriptureProduct, 'unspecifiedScripture', 'totalVerses'))
+                  }
+              )
+              else {},
+          mediums := <str>json_array_unpack(json_get(directScriptureProduct,'mediums')),
+          purposes := <str>json_array_unpack(json_get(directScriptureProduct,'purposes')),
+          steps := <str>json_array_unpack(json_get(directScriptureProduct,'steps')),
+          methodology := <Product::Methodology>json_get(directScriptureProduct,'methodology'),
+          describeCompletion := <str>json_get(directScriptureProduct,'describeCompletion'),
+          placeholderDescription := <str>json_get(directScriptureProduct,'placeholderDescription'),
+          pnpIndex := <int16>json_get(directScriptureProduct,'pnpIndex'),
+          progressTarget := <int16>directScriptureProduct['progressTarget'],
+          progressStepMeasurement := <str>directScriptureProduct['progressStepMeasurement'],
+          totalVerses := <int16>json_get(directScriptureProduct, 'totalVerses'),
+          totalVerseEquivalents := <int32>json_get(directScriptureProduct, 'totalVerseEquivalents')
+        }
+      )
+    )),
+      newProducts := (select products filter .createdAt = datetime_of_statement())
+ select { `Added Scripture Products` := "Project: " ++ newProducts.project.name ++ ', Scripture Range: '
+         ++ newProducts.scripture.label ++ ' ' ++ <str>newProducts.scripture.verses.`start`.chapter ++ ' - ' ++ <str>newProducts.scripture.verses.`end`.chapter, `Total Scripture` := count(newProducts.scripture), `Added UnspecifiedScripture Products` := "Project: "
+         ++ newProducts.project.name ++ ', Unspecified Scripture Book: ' ++ newProducts.unspecifiedScripture.book, `Total UnspecifiedScripture` := count(newProducts.unspecifiedScripture)}
```

Comment:
I won't nitpick on formatting here, as we don't have a way yet to enforce it.
But can you put each of these 4 on their own line.

It may just be me, but I'd simplify this format.
```diff
- Project: Misty Mountains, Scripture Range: Proverbs 1 - 31
+ Misty Mountains: Proverbs 1 - 31
```
```diff
- Project: Misty Mountains, Unspecified Scripture Book: Matthew
+ Misty Mountains: Matthew X verses
```
We are already scoped with a
```
"Added Scripture Products": [
  ...
]
```
which gives enough info. So emitting types just adds clutter.
Thoughts?

Reply:
I toggled back and forth on this and I feel fine with simplifying it.  I just wanted to make sure I had the necessary info output clearly, which I think is still accomplished with your suggestions.